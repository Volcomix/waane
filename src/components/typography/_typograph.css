@import 'https://fonts.googleapis.com/css?family=Roboto:300,400,500&display=swap';

:root,
:host {
  --typeface: var(--w-typeface, 'Roboto', sans-serif);

  --font-light: var(--w-font-light, 300);
  --font-regular: var(--w-font-regular, 400);
  --font-medium: var(--w-font-medium, 500);
}
