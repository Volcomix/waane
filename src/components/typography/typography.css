.typography--headline1 {
  font-family: var(--typeface);
  font-weight: var(--font-light);
  font-size: 96px;
  letter-spacing: -1.5px;
}

.typography--headline2 {
  font-family: var(--typeface);
  font-weight: var(--font-light);
  font-size: 60px;
  letter-spacing: -0.5px;
}

.typography--headline3 {
  font-family: var(--typeface);
  font-weight: var(--font-regular);
  font-size: 48px;
}

.typography--headline4 {
  font-family: var(--typeface);
  font-weight: var(--font-regular);
  font-size: 34px;
  letter-spacing: 0.25px;
}

.typography--headline5 {
  font-family: var(--typeface);
  font-weight: var(--font-regular);
  font-size: 24px;
}

.typography--headline6 {
  font-family: var(--typeface);
  font-weight: var(--font-medium);
  font-size: 20px;
  letter-spacing: 0.15px;
}

.typography--subtitle1 {
  font-family: var(--typeface);
  font-weight: var(--font-regular);
  font-size: 16px;
  letter-spacing: 0.15px;
}

.typography--subtitle2 {
  font-family: var(--typeface);
  font-weight: var(--font-medium);
  font-size: 14px;
  letter-spacing: 0.1px;
}

.typography--body1 {
  font-family: var(--typeface);
  font-weight: var(--font-regular);
  font-size: 16px;
  letter-spacing: 0.5px;
}

.typography--body2 {
  font-family: var(--typeface);
  font-weight: var(--font-regular);
  font-size: 14px;
  letter-spacing: 0.25px;
}

.typography--button {
  font-family: var(--typeface);
  font-weight: var(--font-medium);
  font-size: 14px;
  text-transform: uppercase;
  letter-spacing: 1.25px;
}

.typography--caption {
  font-family: var(--typeface);
  font-weight: var(--font-regular);
  font-size: 12px;
  letter-spacing: 0.4px;
}

.typography--overline {
  font-family: var(--typeface);
  font-weight: var(--font-regular);
  font-size: 10px;
  text-transform: uppercase;
  letter-spacing: 1.5px;
}
