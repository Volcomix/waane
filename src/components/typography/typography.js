import { css } from '../waane-element.js'

const typeface = 'var(--typeface)'
const light = `var(--font-light)`
const regular = `var(--font-regular)`
const medium = `var(--font-medium)`

const typescale = {
  headline1: [light, 96, null, -1.5],
  headline2: [light, 60, null, -0.5],
  headline3: [regular, 48, null, 0],
  headline4: [regular, 34, null, 0.25],
  headline5: [regular, 24, null, 0],
  headline6: [medium, 20, null, 0.15],
  subtitle1: [regular, 16, null, 0.15],
  subtitle2: [medium, 14, null, 0.1],
  body1: [regular, 16, null, 0.5],
  body2: [regular, 14, null, 0.25],
  button: [medium, 14, 'uppercase', 1.25],
  caption: [regular, 12, null, 0.4],
  overline: [regular, 10, 'uppercase', 1.5],
}

export default function typography(name) {
  const [font, size, textTransform, letterSpacing] = typescale[name]
  return css`
    font-size: ${size}px;
    font-family: ${typeface};
    font-weight: ${font};
    text-transform: ${textTransform || 'none'};
    letter-spacing: ${letterSpacing}px;
  `
}
