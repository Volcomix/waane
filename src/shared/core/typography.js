import { css } from './element.js'

const light = 300
const regular = 400
const medium = 500

export const typescale = {
  headline1: [light, 96, null, -1.5],
  headline2: [light, 60, null, -0.5],
  headline3: [regular, 48, null, 0],
  headline4: [regular, 34, null, 0.25],
  headline5: [regular, 24, null, 0],
  headline6: [medium, 20, null, 0.15],
  subtitle1: [regular, 16, null, 0.15],
  subtitle2: [medium, 14, null, 0.1],
  body1: [regular, 16, null, 0.5],
  body2: [regular, 14, null, 0.25],
  button: [medium, 14, 'uppercase', 1.25],
  caption: [regular, 12, null, 0.4],
  overline: [regular, 10, 'uppercase', 1.5],
}

/**
 * @param {keyof typeof typescale} name
 */
export default function typography(name) {
  const [weight, size, textTransform, letterSpacing] = typescale[name]
  return css`
    font-size: ${size}px;
    font-weight: ${weight};
    text-transform: ${textTransform || 'none'};
    letter-spacing: ${letterSpacing}px;
  `
}

const style = document.createElement('style')
style.textContent = css`
  @import url('https://fonts.googleapis.com/css2?family=Roboto:wght@300;400;500&display=swap');

  body {
    font-family: 'Roboto', sans-serif;
    ${typography('body1')}
  }
`
document.head.appendChild(style)
