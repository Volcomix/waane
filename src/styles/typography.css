@import url('https://fonts.googleapis.com/css?family=Roboto:300,400,500');

body {
  font-family: 'Roboto', sans-serif;
}

.headline1,
h1 {
  font-size: 96px;
  font-weight: 300;
}
.headline2,
h2 {
  font-size: 60px;
  font-weight: 300;
}
.headline3,
h3 {
  font-size: 48px;
  font-weight: 400;
}
.headline4,
h4 {
  font-size: 34px;
  font-weight: 400;
}
.headline5,
h5 {
  font-size: 24px;
  font-weight: 400;
}
.headline6,
h6 {
  font-size: 20px;
  font-weight: 500;
}
.subtitle1 {
  font-size: 16px;
  font-weight: 400;
}
.subtitle2 {
  font-size: 14px;
  font-weight: 500;
}
.body1 {
  font-size: 16px;
  font-weight: 400;
}
.body2 {
  font-size: 14px;
  font-weight: 400;
}
.button {
  font-size: 14px;
  font-weight: 500;
  text-transform: uppercase;
}
.caption {
  font-size: 12px;
  font-weight: 400;
}
.overline {
  font-size: 14px;
  font-weight: 400;
  text-transform: uppercase;
}