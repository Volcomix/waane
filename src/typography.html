<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="utf-8" />
    <title>Typography</title>
    <meta name="description" content="Typography test page" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <link
      href="https://fonts.googleapis.com/css?family=Roboto:300,400,500&display=swap"
      rel="stylesheet"
    />
    <style>
      body {
        display: flex;
        flex-direction: column;
      }

      h1,
      h2,
      h3,
      h4,
      h5,
      h6,
      p {
        margin: 0;
      }
    </style>
  </head>
  <body>
    <h1 class="headline1">Headline 1</h1>
    <h2 class="headline2">Headline 2</h2>
    <h3 class="headline3">Headline 3</h3>
    <h4 class="headline4">Headline 4</h4>
    <h5 class="headline5">Headline 5</h5>
    <h6 class="headline6">Headline 6</h6>
    <h6 class="subtitle1">Subtitle 1</h6>
    <h6 class="subtitle2">Subtitle 2</h6>
    <p class="body1">Body 1</p>
    <p class="body2">Body 2</p>
    <span class="button">Button</span>
    <span class="caption">Caption</span>
    <span class="overline">Overline</span>

    <script type="module">
      import { css } from './components/waane-element.js'
      import variables from './components/variables.js'
      import typography from './components/typography/typography.js'

      const variablesStyle = variables(':root')
      document.head.insertAdjacentHTML('beforeend', variablesStyle)

      const style = document.createElement('style')
      const content = []
      document.body.querySelectorAll('[class]').forEach(({ className }) => {
        content.push(css`
          .${className} {
            ${typography(className)}
          }
        `)
      })
      style.textContent = content.join('')
      document.head.appendChild(style)
    </script>
  </body>
</html>
