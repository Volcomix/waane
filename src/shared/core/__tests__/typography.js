import { expect, test } from '@jest/globals'
import typography, { typescale } from '../typography'

Object.keys(typescale).forEach((/** @type {keyof typeof typescale} */ name) => {
  test(`returns styles for ${name}`, () => {
    expect(typography(name)).toMatchSnapshot()
  })
})
