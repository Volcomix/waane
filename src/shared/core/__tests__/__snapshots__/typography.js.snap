// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`returns styles for body1 1`] = `
"
    font-size: 16px;
    font-weight: 400;
    text-transform: none;
    letter-spacing: 0.5px;
  "
`;

exports[`returns styles for body2 1`] = `
"
    font-size: 14px;
    font-weight: 400;
    text-transform: none;
    letter-spacing: 0.25px;
  "
`;

exports[`returns styles for button 1`] = `
"
    font-size: 14px;
    font-weight: 500;
    text-transform: uppercase;
    letter-spacing: 1.25px;
  "
`;

exports[`returns styles for caption 1`] = `
"
    font-size: 12px;
    font-weight: 400;
    text-transform: none;
    letter-spacing: 0.4px;
  "
`;

exports[`returns styles for headline1 1`] = `
"
    font-size: 96px;
    font-weight: 300;
    text-transform: none;
    letter-spacing: -1.5px;
  "
`;

exports[`returns styles for headline2 1`] = `
"
    font-size: 60px;
    font-weight: 300;
    text-transform: none;
    letter-spacing: -0.5px;
  "
`;

exports[`returns styles for headline3 1`] = `
"
    font-size: 48px;
    font-weight: 400;
    text-transform: none;
    letter-spacing: 0px;
  "
`;

exports[`returns styles for headline4 1`] = `
"
    font-size: 34px;
    font-weight: 400;
    text-transform: none;
    letter-spacing: 0.25px;
  "
`;

exports[`returns styles for headline5 1`] = `
"
    font-size: 24px;
    font-weight: 400;
    text-transform: none;
    letter-spacing: 0px;
  "
`;

exports[`returns styles for headline6 1`] = `
"
    font-size: 20px;
    font-weight: 500;
    text-transform: none;
    letter-spacing: 0.15px;
  "
`;

exports[`returns styles for overline 1`] = `
"
    font-size: 10px;
    font-weight: 400;
    text-transform: uppercase;
    letter-spacing: 1.5px;
  "
`;

exports[`returns styles for subtitle1 1`] = `
"
    font-size: 16px;
    font-weight: 400;
    text-transform: none;
    letter-spacing: 0.15px;
  "
`;

exports[`returns styles for subtitle2 1`] = `
"
    font-size: 14px;
    font-weight: 500;
    text-transform: none;
    letter-spacing: 0.1px;
  "
`;
